src/utils/registerDockerContextStatusBarItems.ts

```diff
@@ -0,0 +1,55 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IActionContext, registerEvent } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { ext } from "../extensionVariables";
+import { ListContextItem } from '../runtimes/docker/contracts/ContainerClient';
+
+const dockerContextStatusBarSetting = 'contexts.showInStatusBar';
+
+export function registerDockerContextStatusBarEvent(ctx: vscode.ExtensionContext): void {
+    // Register an event to watch for changes to config, reconfigure if needed
+    registerEvent('docker.command.changed', vscode.workspace.onDidChangeConfiguration, (actionContext: IActionContext, e: vscode.ConfigurationChangeEvent) => {
+
+        actionContext.telemetry.suppressAll = true;
+        actionContext.errorHandling.suppressDisplay = true;
+
+        if (e.affectsConfiguration('docker.contexts.showInStatusBar')) {
+            // Don't wait
+            void registerDockerContextStatusBarItems();
+        }
+    });
+
+    ctx.subscriptions.push(
+        ext.dockerContextStatusBarItem,
+        ext.runtimeManager.contextManager.onContextChanged(registerDockerContextStatusBarItems)
+    );
+}
+
+export async function registerDockerContextStatusBarItems() {
+
+    const config = vscode.workspace.getConfiguration('docker');
+    let currentDockerContext: ListContextItem | undefined;
+    ext.dockerContextStatusBarItem?.dispose();
+
+    // if dockerContextStatusBarItem is created, then we dispose
+    if (!config.get(dockerContextStatusBarSetting, false) ||
+        !(currentDockerContext = await ext.runtimeManager.contextManager.getCurrentContext())) { // Intentional assignment and boolean check
+        return;
+    }
+
+    const dockerContextUseCommand = 'vscode-docker.contexts.use';
+    const currentContextName = `Context: ${currentDockerContext.name}`;
+    currentDockerContext = await ext.runtimeManager.contextManager.getCurrentContext();
```

Comment:
This line is redundant; the above `!(currentDockerContext = await ext.runtimeManager.contextManager.getCurrentContext()))` will have already set `currentDockerContext`

```suggestion
```